clang/runtime/dpct-rt/include/dpct/dpl_extras/dpcpp_extensions.h

```diff
@@ -460,6 +460,49 @@ class radix_sort {
         break;
     }
 
+#pragma unroll
+    for (int i = 0; i < VALUES_PER_THREAD; ++i) {
+      unsigned_keys[i] = detail::traits<T>::twiddle_out(unsigned_keys[i]);
+    }
+  }
+
+  template <typename Item>
+  __dpct_inline__ void
+  sortBlockedToStriped(const Item &item, T (&keys)[VALUES_PER_THREAD],
+                       int begin_bit = 0, int end_bit = 8 * sizeof(T)) {
+
+    uint32_t(&unsigned_keys)[VALUES_PER_THREAD] =
+        reinterpret_cast<uint32_t(&)[VALUES_PER_THREAD]>(keys);
+
+#pragma unroll
+    for (int i = 0; i < VALUES_PER_THREAD; ++i) {
+      unsigned_keys[i] = detail::traits<T>::twiddle_in(unsigned_keys[i]);
+    }
+
+    while (true) {
```

Comment:
Is it possible to reorganize this loop a bit to have an explicit condition instead of `while (true)` with a break in the middle of the loop? It makes the code a bit more clear in my opinion.

Reply:
Yeah I thought so too, but I am not sure how to modify this. Even in nv's code, it starts off with a infinite looping with break inside.

Reply:
Its just iterating through the radix from `begin_bit` to `end_bit` with a branch for the final iteration.  Should be doable with a for loop and an if/else statement.